src/KingSystem/Map/mapObject.cpp

```diff
@@ -1,7 +1,373 @@
 #include "KingSystem/Map/mapObject.h"
+#include "KingSystem/ActorSystem/actBaseProcMgr.h"
+#include "KingSystem/Utils/Debug.h"
 
 namespace ksys::map {
 
 Object::Object() = default;
 
+Object::~Object() {
+    if (mLinkData != nullptr) {
+        mLinkData->release(this, true);
+        mLinkData = nullptr;
+    }
 }
+
+void Object::free() {
+    mId = 0;
+    _10 = nullptr;
+    _18 = nullptr;
+    _20 = nullptr;
+
+    if (mProc)
+        mProc = nullptr;
+
+    // TODO: figure out what this is
+    mFlags0.reset(Flag0::Freed);
+
+    mHardModeFlags.reset(HardModeFlag::ActorCreated);
+
+    if (mFlags0.isOn(Flag0::_80000))
+        mFlags0.set(Flag0::_40000);
+    else
+        mFlags0.set(Flag0::_8);
+
+    if (mLinkData)
+        mLinkData->field_57 = 0;
+}
+
+// NON_MATCHING: functionally equivalent; CSEL instruction flipped
+bool Object::getActorWithAccessor(act::ActorLinkConstDataAccess& accessor) const {
+    act::BaseProcMgr* instance = act::BaseProcMgr::instance();
+
+    if (instance == nullptr || instance->isHighPriorityThread() == false || mProc == nullptr)
+        return accessor.acquire(nullptr);
+
+    if (mProc->isInit() || mProc->isCalc())
+        return accessor.acquire(mProc);
+
+    return accessor.acquire(nullptr);
+}
+
+act::Actor* Object::getActor() const {
+    auto* actor = sead::DynamicCast<act::Actor>(mProc);
+    if (actor == nullptr)
```

Comment:
Is this actually necessary? I think `return sead::DynamicCast<act::Actor>(mProc);` should suffice